frontend/app/src/screens/StakeScreen/components/PanelVoting/components/EpochInitiativesTable/components/TableBody/hooks/useActiveInitiatives/useActiveInitiatives.tsx

```diff
@@ -0,0 +1,41 @@
+import { useMemo } from "react";
+import { useVotingStateContext } from "@/src/screens/StakeScreen/components/PanelVoting/providers/PanelVotingProvider/hooks";
+import { isInitiativeStatusActive } from "@/src/screens/StakeScreen/utils";
+
+export const useActiveInitiatives = () => {
+  const { initiativesData, initiativesStatesData, voteAllocations } =
+    useVotingStateContext();
+
+  return useMemo(() => {
+    return (initiativesData || [])
+      .filter((initiative) => {
+        const status =
+          initiativesStatesData?.[initiative.address]?.status ?? "nonexistent";
+
+        return (
+          isInitiativeStatusActive(status) ||
+          Boolean(voteAllocations[initiative.address])
+        );
+      })
+      .sort((a, b) => {
+        // 1. sort by allocation
+        const allocationA = voteAllocations[a.address];
+        const allocationB = voteAllocations[b.address];
+        if (allocationA && !allocationB) return -1;
+        if (!allocationA && allocationB) return 1;
+
+        // 2. sort by status
+        const statusA =
+          initiativesStatesData?.[a.address]?.status ?? "nonexistent";
+        const statusB =
+          initiativesStatesData?.[b.address]?.status ?? "nonexistent";
+        const isActiveA = isInitiativeStatusActive(statusA);
+        const isActiveB = isInitiativeStatusActive(statusB);
+
+        if (isActiveA && !isActiveB) return -1;
+        if (!isActiveA && isActiveB) return 1;
+
+        return 0;
+      });
+  }, [initiativesData, initiativesStatesData]);
```

Comment:
`voteAllocations` is missing from the dependency list. As a result, initiatives can be ordered incorrectly after switching accounts in the wallet.